### PR TITLE
phonon-backend-vlc: use qt5 by default

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

`minitube` is currently broken transitively due to the broken
`phonon-backend-qt4`: https://hydra.nixos.org/build/76523277

Although QT4 is fairly old, this package is still built with `qt4` ATM,
however QT5 is available as well. After this change, QT5 will be built
by default and in case anybody requires legacy QT4 it has to be enabled
explicitly like this:

```
with import <nixpkgs> { };
phonon-backend-vlc.override { withQt4 = true; }
```

Now the QT5-only build can be used (which fixes `minitube`) and there
are no confusions anymore with the build dependencies. Previously
`phonon-backend-vlc` and `libsForQt5.phonon-backend-vlc` used `qt4` by
default which was likely responsible for broken `minitube`.

The only package which uses `phonon-backend-vlc` as well is `tomahawk`
which is currently unmaintained in `nixpkgs` (see c76ad476b0245c474c057b451ba6fec18e7f81f1)
and broken since March (https://hydra.nixos.org/build/71873808).
FOr now I've marked it as broken as the last *stable* release is from
2015 (see https://github.com/tomahawk-player/tomahawk/releases/tag/0.8.4).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by xeji** [comment]
@GrahamcOfBorg build libsForQt5.phonon-backend-vlc minitube

**Comment by xeji** [comment]
aarch64 failure: dependency `live555` has been broken for a while, unrelated to this PR.

File: pkgs/applications/audio/tomahawk/default.nix [hunk 1]
```diff
@@ -51,5 +51,6 @@ stdenv.mkDerivation rec {
     homepage = http://tomahawk-player.org/;
     license = licenses.gpl3Plus;
     platforms = platforms.all;
+    broken = true; # 2018-06-25
```
Comment (xeji) [inline pkgs/applications/audio/tomahawk/default.nix]: How is this related to phonon-backend-vlc ?
Also, please put this in a separate commit for a clean history.

Comment (Ma27) [reply pkgs/applications/audio/tomahawk/default.nix]: `tomahawk` was the only other package which depends on `phonon-backend-vlc`.
However it doesn't build anymore for some time (https://hydra.nixos.org/build/71873808) and I don't think it's worth to try making it compatible with my change as it's now unmaintained for quite a while according to c76ad476b0245c474c057b451ba6fec18e7f81f1)

**Comment by Ma27** [comment]
@xeji created now two commits with a description and needed links to the affected Hydra builds. Is this fine for you now? %)

**Comment by xeji** [comment]
Thanks :+1:

**Comment by xeji** [comment]
@GrahamcOfBorg build libsForQt5.phonon-backend-vlc minitube
